ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -75,7 +76,7 @@ forward the
    *
    * \return True if the element is displacement-based, false otherwise.
    */
-  bool isDisplacementBased() const { return std::holds_alternative<std::monostate>(easVariant_); }
+  bool isDisplacementBased() const { return std::holds_alternative<EASBaseType>(easVariant_.var); }
```

Comment:
Don't call here the member variable, since it is an impl detail. Write a function for easVariant_ which you call here. Then you don't need `EASBaseType` here at all.

Reply:
Inside the function you can simply do this by 

```cpp
bool isNoOp() const {
    return getNumberOfEASParameters()==0;
  }

Reply:
Or maybe a better name